### PR TITLE
Fix PredictionContext merge() typo

### PR DESCRIPTION
**Pull request by pboyer**
cc @sridharxp

This is the fix in #60 + another bug fix.

### CODE REVIEW

**Comment by pboyer** [comment]
After this fix:

```
Results :
Failed tests:   testPositionAdjustingLexer(org.antlr.v4.test.runtime.go.TestLexerExec): expected:<...ns',<6>,1:0](..)
  testZeroLengthToken(org.antlr.v4.test.runtime.go.TestLexerExec): expected:<[@0,0:[4=''xxx'',<1>,1:0](..)
  testBasic(org.antlr.v4.test.runtime.go.TestListeners): expected:<(a 1 2)(..)
  testReturnValueAndActionsList1_2(org.antlr.v4.test.runtime.go.TestLeftRecursion): expected:<...pr (expr a) , (expr [c) >> (expr x])) <EOF>)(..)
  testReturnValueAndActionsList1_4(org.antlr.v4.test.runtime.go.TestLeftRecursion): expected:<...* (expr b)) , (expr [c) , (expr (expr x) * (expr y)) >> (expr r)]) <EOF>)(..)
Tests in error: 
  testInvalidATNStateRemoval(org.antlr.v4.test.runtime.go.TestParserErrors): test timed out after 60000 milliseconds
Tests run: 314, Failures: 5, Errors: 1, Skipped: 0
```

With #59 we're down to 5 tests.

File: runtime/Go/antlr/prediction_context.go [hunk 1]
```diff
@@ -615,8 +615,8 @@ func mergeArrays(a, b *ArrayPredictionContext, rootIsWildcard bool, mergeCache *
 	var j = 0 // walks b
 	var k = 0 // walks target M array
 
-	var mergedReturnStates = make([]int, 0)
-	var mergedParents = make([]PredictionContext, 0)
+	var mergedReturnStates = make([]int, len(a.returnStates) + len(b.returnStates))
+	var mergedParents = make([]PredictionContext, len(a.returnStates) + len(b.returnStates))
```
Comment (pboyer) [inline runtime/Go/antlr/prediction_context.go:619]: The typo error revealed this mistake.

**Comment by pboyer** [comment]
@willfaught PTAL.

**Comment by willfaught** [comment]
Fantastic! LGTM

**Pull request merged**